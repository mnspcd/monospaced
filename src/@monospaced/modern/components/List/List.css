.mdrn .List {
  list-style: none;
  margin-bottom: 0;
  margin-top: 0;
  max-width: var(--measure);
  padding-left: 0;
}

.mdrn .List--ordered {
  counter-reset: list;
}

.mdrn .List li {
  padding-left: 1.5em;
  position: relative;
}

.mdrn .List li::before {
  left: 0;
  position: absolute;
  text-align: center;
  width: 1.5em;
}

.mdrn .List > li::before {
  content: "â€¢";
}

.mdrn .List--ordered > li::before {
  content: counter(list) ".";
  counter-increment: list;
  font-family: var(--font-small-caps);
}
