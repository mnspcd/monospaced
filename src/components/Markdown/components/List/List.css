.List {
  list-style: none;
  margin-bottom: 0;
  margin-top: 0;
  max-width: var(--measure);
  padding-left: 0;
}

.List--ordered {
  counter-reset: list;
}

.List li {
  padding-left: 1.5em;
  position: relative;
}

.List li::before {
  left: 0;
  position: absolute;
  text-align: center;
  width: 1.5em;
}

.List > li::before {
  content: "â€¢";
}

.List--ordered > li::before {
  content: counter(list) ".";
  counter-increment: list;
  font-family: var(--font-small-caps);
}
