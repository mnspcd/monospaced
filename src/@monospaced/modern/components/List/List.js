import classNames from "classnames";
import PropTypes from "prop-types";
import React from "react";

import "./List.css";

/**
 * Use `List` to typset text in lists.
 */
const List = ({ children, ordered = false }) => {
  let Element = "ul";

  if (ordered) {
    Element = "ol";
  }

  return (
    <Element className={classNames({ List: true, [`List--ordered`]: ordered })}>
      {children}
    </Element>
  );
};

List.propTypes = {
  /**
   * HTML list item elements containing text to display in the List.
   */
  children: PropTypes.node.isRequired,

  /**
   * Ordered variant of List.
   */
  ordered: PropTypes.bool,
};

List.Ordered = ({ children }) => <List ordered>{children}</List>;

List.Ordered.displayName = "List.Ordered";

List.Ordered.propTypes = {
  children: PropTypes.node.isRequired,
};

export default List;
