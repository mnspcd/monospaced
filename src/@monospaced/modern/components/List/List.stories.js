import { boolean, number, text } from "@storybook/addon-knobs";
import React from "react";

import List from "./List";

export default {
  component: List,
  title: "Components/List",
};

export const basic = () => (
  <List>
    <li>Lorem ipsum dolor sit amet, consectetur adipiscing elit</li>
    <li>Lorem ipsum dolor sit amet, consectetur adipiscing elit</li>
    <li>Lorem ipsum dolor sit amet, consectetur adipiscing elit</li>
  </List>
);

export const ordered = () => (
  <List ordered>
    <li>Lorem ipsum dolor sit amet, consectetur adipiscing elit</li>
    <li>Lorem ipsum dolor sit amet, consectetur adipiscing elit</li>
    <li>Lorem ipsum dolor sit amet, consectetur adipiscing elit</li>
  </List>
);

export const knobs = () => {
  const numberOfItems = number(
    "number of items",
    3,
    {
      range: true,
      min: 1,
      max: 99,
    },
    "Story",
  );
  const isOrdered = boolean("ordered", false, "List");
  const itemText = text(
    "item text",
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
    "List",
  );

  return (
    <List ordered={isOrdered}>
      {[...Array(numberOfItems).keys()].map(i => (
        <li key={i}>{itemText}</li>
      ))}
    </List>
  );
};

knobs.story = { parameters: { docs: { disable: true } } };
