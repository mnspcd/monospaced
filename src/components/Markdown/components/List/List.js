import PropTypes from "prop-types";
import React from "react";
import classNames from "classnames";

import "./List.css";

const List = ({ children, ordered }) => {
  let Element = "ul";

  if (ordered) {
    Element = "ol";
  }

  return (
    <Element className={classNames({ List: true, [`List--ordered`]: ordered })}>
      {children}
    </Element>
  );
};

List.Ordered = ({ children }) => <List ordered>{children}</List>;

List.propTypes = {
  children: PropTypes.node.isRequired,
  ordered: PropTypes.bool,
};

export default List;
