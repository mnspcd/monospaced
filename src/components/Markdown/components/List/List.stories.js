import React from "react";
import { boolean, number, text } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import List from "./List";

storiesOf("Components/Markdown/List", module).add("Component", () => {
  const numberOfItems = number(
    "number of items",
    3,
    {
      range: true,
      min: 1,
      max: 99,
    },
    "Story",
  );
  const ordered = boolean("ordered", false, "List");
  const itemText = text(
    "item text",
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
    "List",
  );

  return (
    <List ordered={ordered}>
      {[...Array(numberOfItems).keys()].map(i => (
        <li key={i}>{itemText}</li>
      ))}
    </List>
  );
});
