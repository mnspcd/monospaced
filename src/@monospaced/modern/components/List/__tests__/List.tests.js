/* eslint-env jest */
import { render } from "@testing-library/react";
import React from "react";

import List from "..";

describe("List component", () => {
  it("should render correctly", () => {
    const { container } = render(<List>children</List>);

    expect(container).toMatchSnapshot();
  });

  it("should render an ordered list correctly", () => {
    const { container } = render(<List.Ordered>children</List.Ordered>);

    expect(container).toMatchSnapshot();
  });
});
