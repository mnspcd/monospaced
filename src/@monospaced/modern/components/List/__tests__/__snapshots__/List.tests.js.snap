// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should render an ordered list correctly 1`] = `
<div>
  <ol
    class="List List--ordered"
  >
    children
  </ol>
</div>
`;

exports[`List component should render correctly 1`] = `
<div>
  <ul
    class="List"
  >
    children
  </ul>
</div>
`;
